pkg/cloudinstances/cloud_instance.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinstances
+
+import v1 "k8s.io/api/core/v1"
+
+// CloudInstanceStatusDetached means the instance needs update and has been detached
+const CloudInstanceStatusDetached = "Detached"
+
+// CloudInstanceStatusNeedsUpdate means the instance needs update
+const CloudInstanceStatusNeedsUpdate = "NeedsUpdate"
+
+// CloudInstanceStatusReady means the instance has joined the cluster and is up to date
+const CloudInstanceStatusReady = "Ready"
+
+// CloudInstanceStatusNotJoined means the instance has not yet joined the cluster
+const CloudInstanceStatusNotJoined = "NotJoined"
```

Comment:
I suppose you are right. It gets a bit awkward either way. If we were to remove this, it would mean that we'd have ready nodes that have not yet joined the cluster. But not being able to detect detached nodes that are absent from the cluster would be worse.
Clould be that "ready" is not the right term for this status. Maybe it should be "UpToDate" or "Running" or something is better.

Reply:
I think you're right that "Ready" is the wrong term for this. I believe with one of your other changes it could be in this state even when it's pending.

I think I have a slight preference for "UpToDate". Another possibility might be "Current".

Also, consider whether `instancegroups.AdjustNeedUpdate()` needs to change the `Status` field.